docker-build-system/zombodb-build-alpine-3.12/Dockerfile

```diff
@@ -0,0 +1,75 @@
+#
+# Copyright 2018-2020 ZomboDB, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+FROM rust:1.47-alpine3.12 as zombodb-extension
+
+LABEL maintainer="ZomboDB, LLC (zombodb@gmail.com)"
+
+ARG USER=docker
+ARG UID=1000
+ARG GID=1000
+RUN adduser --uid=${UID} -D ${USER}
+
+RUN apk add --no-cache \
+	musl-dev \
+	openssl-dev \
+	make \
+	bison \
+	coreutils \
+	dpkg-dev dpkg \
+	flex \
+	gcc \
+	libc-dev \
+	libedit-dev \
+	libxml2-dev \
+	libxslt-dev \
+	linux-headers \
+	make \
+	openssl-dev \
+	perl-utils \
+	perl-ipc-run \
+	util-linux-dev \
+	zlib-dev \
+	icu-dev
+
+USER ${UID}:${GID}
+WORKDIR /home/${USER}
+
+RUN cargo install cargo-pgx
+COPY build_pg.sh build_pg.sh
+
+ENV PG_10=10.15
+RUN wget https://ftp.postgresql.org/pub/source/v${PG_10}/postgresql-${PG_10}.tar.gz -qO /home/${USER}/postgresql-10.tar.gz
+RUN sh ./build_pg.sh 10
+
+ENV PG_11=11.10
+RUN wget https://ftp.postgresql.org/pub/source/v${PG_11}/postgresql-${PG_11}.tar.gz -qO /home/${USER}/postgresql-11.tar.gz
+RUN sh ./build_pg.sh 11
+
+ENV PG_12=12.5
+RUN wget https://ftp.postgresql.org/pub/source/v${PG_12}/postgresql-${PG_12}.tar.gz -qO /home/${USER}/postgresql-12.tar.gz
+RUN sh ./build_pg.sh 12
+
+ENV PG_13=13.1
+RUN wget https://ftp.postgresql.org/pub/source/v${PG_13}/postgresql-${PG_13}.tar.gz -qO /home/${USER}/postgresql-13.tar.gz
+RUN sh ./build_pg.sh 13
+
+RUN cargo pgx init \
+	--pg10=/home/${USER}/postgresql-10/build/bin/pg_config \
+	--pg11=/home/${USER}/postgresql-11/build/bin/pg_config \
+	--pg12=/home/${USER}/postgresql-12/build/bin/pg_config \
+	--pg13=/home/${USER}/postgresql-13/build/bin/pg_config
+
+ENV RUSTFLAGS="-C target-feature=-crt-static"
```

Comment:
I understand what this does, but I do **not** understand why it's necessary.

ZomboDB, through its `Cargo.toml` file, is setup to build a cdylib, so I don't see why this is necessary.

What version of Rust is alpine installing?  I don't have a minimum Rust version requirement, but maybe I need to?

Reply:
the image is rust:1.47-alpine3.12 si that’s rust version 1.47

once I can build the extension in this image I’ll try with and without it

Reply:
so I tried with the current Dockerfile with huste this line removed I have this error:

```
[...]
  Downloaded encoding_rs v0.8.26
error: cannot produce cdylib for `zombodb v3000.0.0-alpha1 (/build/zombodb)` as the target `x86_64-unknown-linux-musl` does not support these crate types
     [error] failed to build extension
+ exit 1
```